tools/dump-dataset.py

```diff
@@ -222,7 +222,7 @@ def init(config_str, config_dataset, verbosity):
     config.set("train", dataset_dict)
   elif config_dataset and config_dataset != "train":
     print("Use dataset %r from config." % config_dataset)
-    config.set("train", "config:%s" % config_dataset)
+    config.set("train", config.typed_value(config_dataset, None))
```

Comment:
This is not the same.

We also allow to pass e.g. `--dataset 'get_dataset("train")'`, which we commonly use for configs where the dataset is given by such a `get_dataset` function. The given string is then `eval`d in the context of the config.

With your change, this breaks.

Reply:
I see. Having defined `dev` as a dict doesn't work, though. Maybe we should make sure we don't set `dev` to `None` if we specify `--dataset dev`.

Reply:
> I see. Having defined `dev` as a dict doesn't work, though. Maybe we should make sure we don't set `dev` to `None` if we specify `--dataset dev`.

I would avoid heuristics. There could be other ways how this fails then. But my other suggestion (replacing `rnn.init_data`) is anyway cleaner and simpler and should not have such problems. See `dump-forward-stats.py` as an example.